clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchFieldMapper.scala

```diff
@@ -32,34 +42,18 @@ private[dataaccess] object ElasticsearchFieldMapper
 
     override def stringToDefinition(fieldType: Type): String => FieldDefinition = {
       fieldType match {
-        case tpe if tpe =:= typeOf[Boolean]           => booleanField
-        case tpe if tpe =:= typeOf[Int]               => intField
-        case tpe if tpe =:= typeOf[Long]              => longField
-        case tpe if tpe =:= typeOf[Float]             => floatField
-        case tpe if tpe =:= typeOf[Double]            => doubleField
-        case tpe if tpe =:= typeOf[String]            => keywordField
-        case tpe if tpe =:= typeOf[Symbol]            => keywordField
-        case tpe if tpe <:< typeOf[EnumEntry]         => keywordField
-        case tpe if tpe =:= typeOf[OffsetDateTime]    => dateField
-        case tpe if tpe =:= typeOf[UpsertId]          => keywordField
-        case tpe if tpe =:= typeOf[Option[Boolean]]   => booleanField
-        case tpe if tpe =:= typeOf[Option[Int]]       => intField
-        case tpe if tpe =:= typeOf[Option[Long]]      => longField
-        case tpe if tpe =:= typeOf[Option[Float]]     => floatField
-        case tpe if tpe =:= typeOf[Option[Double]]    => doubleField
-        case tpe if tpe =:= typeOf[Option[String]]    => keywordField
-        case tpe if tpe =:= typeOf[Option[Symbol]]    => keywordField
-        case tpe if tpe <:< typeOf[Option[EnumEntry]] => keywordField
-        case tpe if tpe =:= typeOf[Option[UUID]]      => keywordField
-        case tpe if tpe =:= typeOf[Option[URI]]       => keywordField
-        case tpe if tpe =:= typeOf[Option[OffsetDateTime]] =>
-          dateField
-        /*
-         * Elasticsearch has no notion of an array field; instead, any
-         * field can hold 1+ values of the same type.
-         */
-        case tpe if tpe <:< typeOf[Option[Seq[String]]] => keywordField
-        case tpe if tpe <:< typeOf[Option[Seq[URI]]]    => keywordField
+        case tpe if is[Boolean](tpe)        => booleanField
+        case tpe if is[Int](tpe)            => intField
+        case tpe if is[Long](tpe)           => longField
+        case tpe if is[Float](tpe)          => floatField
+        case tpe if is[Double](tpe)         => doubleField
+        case tpe if is[String](tpe)         => keywordField
+        case tpe if is[Symbol](tpe)         => keywordField
+        case tpe if is[EnumEntry](tpe)      => keywordField
+        case tpe if is[OffsetDateTime](tpe) => dateField
```

Comment:
(Since I made another comment) Nitpick: can you swap this line and the `EnumEntry` line, so all the `keywordField`s are together?